credentials/credentials.go

```diff
@@ -77,23 +111,81 @@ func GetNewRelicLicenseKey(ctx context.Context, conf *config.Configuration) (str
 		return conf.LicenseKey, nil
 	}
 
+	var err error
+	var licenseKey string
+
 	secretId := getLicenseKeySecretId(conf)
+	if secretId != "" {
+		licenseKey, err = tryLicenseKeyFromSecret(ctx, secretId)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	parameterName := getLicenseKeyParameterName(conf)
+	if parameterName != "" {
+		licenseKey, err = tryLicenseKeyFromParameter(ctx, parameterName)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	envLicenseKey, found := os.LookupEnv(defaultSecretId)
+	if found {
+		return envLicenseKey, nil
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	util.Logln("No configured license key found, attempting fallbacks")
+
+	licenseKey, err = tryLicenseKeyFromSecret(ctx, defaultSecretId)
+	if err == nil {
+		return licenseKey, nil
+	}
+
+	licenseKey, err = tryLicenseKeyFromParameter(ctx, defaultSecretId)
+	if err == nil {
+		return licenseKey, nil
+	}
+
+	return "", fmt.Errorf("No license key configured")
+}
+
+func tryLicenseKeyFromSecret(ctx context.Context, secretId string) (string, error) {
```

Comment:
Should this be used inside `IsSecretConfigured`?

Reply:
I think there's a distinction between checking if the secret value is configured, and checking if the secret value is properly formatted and can be decoded. `IsSecretConfigured` just does the former, so it doesn't need to call `decodeLicenseKey`.